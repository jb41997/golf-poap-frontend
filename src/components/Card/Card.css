.card--container {
  display: flex;
  flex-direction: column;
  height: 17rem;
  width: 17rem;
  align-items: center;
  background: linear-gradient(
    0deg,
    rgba(64, 16, 109, 0.4),
    rgba(64, 16, 109, 1)
  );
  box-shadow: 0px 0px 5px var(--primeColor);
  color: white;
  border-radius: 1rem;
  padding: 0.5rem;
}

.wrapper a {
  text-decoration: none;
}

.card--body hr {
  width: 100%;
}

h3 {
  margin: 0;
}

.card--details {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  border-top: 2px solid var(--yellowColor);
  padding-top: 1rem;
}

.card--header img {
  height: 10rem;
  width: 10rem;
  object-fit: cover;
  border-radius: 50%;
}

.card--body {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  width: 100%;
  height: 100%;
  margin-top: 0.5rem;
  text-align: center;
}
