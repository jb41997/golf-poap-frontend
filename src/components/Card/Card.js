import "./Card.css";
import PropTypes from "prop-types";

const Card = (props) => {
  return (
    <div className="wrapper">
      <a
        href={
          props.tokenid
            ? `https://app.poap.xyz/token/${props.tokenid}`
            : `https://poap.gallery/event/${props.id}`
        }
        rel="noreferrer"
        target="_blank"
      >
        <div className="card--container">
          <div className="card--header">
            <img src={props.img} alt={props.name} />
          </div>
          <div className="card--body">
            <h3 title={props.name}>
              {props.name.length > 35
                ? props.name.substr(0, 34) + "\u2026"
                : props.name}
            </h3>
            <div className="card--details">
              <div title="Event ID" className="event--id">
                #{props.id}
              </div>
              <div title="End Date" className="event--end">
                {props.end}
              </div>
            </div>
          </div>
        </div>
      </a>
    </div>
  );
};

Card.propTypes = {
  tokenid: PropTypes.string,
  id: PropTypes.string,
  img: PropTypes.string,
  name: PropTypes.string,
  end: PropTypes.string,
};

export default Card;
