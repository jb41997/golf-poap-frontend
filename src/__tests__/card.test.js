import { render, screen, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom'
import Card from '../components/Card/Card'

afterEach(() => {
  cleanup()
})

test('should render card component', () => {
  const testProps = {
    tokenid: 'tokenid-1',
    id: '0',
    img: 'https://picsum.photos/200/200',
    name: 'test-image',
    end: '10-2022',
  }

  render(
    <Card
      tokenid={testProps.tokenid}
      id={testProps.id}
      img={testProps.img}
      name={testProps.name}
      end={testProps.end}
    />
  )
  const cardElement = screen.getByTestId(`card-${testProps.id}`)
  expect(cardElement).toBeInTheDocument()
  expect(cardElement).toHaveTextContent('test-image')
})

test('matches snapshot', () => {
  const testProps = {
    tokenid: 'tokenid-1',
    id: '0',
    img: 'https://picsum.photos/200/200',
    name: 'test-image',
    end: '10-2022',
  }

  const tree = renderer
    .create(
      <Card
        tokenid={testProps.tokenid}
        id={testProps.id}
        img={testProps.img}
        name={testProps.name}
        end={testProps.end}
      />
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
