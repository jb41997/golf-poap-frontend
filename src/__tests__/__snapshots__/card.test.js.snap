// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="wrapper"
  data-testid="card-0"
>
  <a
    href="https://app.poap.xyz/token/tokenid-1"
    rel="noreferrer"
    target="_blank"
  >
    <div
      className="card--container"
    >
      <div
        className="card--header"
      >
        <img
          alt="test-image"
          src="https://picsum.photos/200/200"
        />
      </div>
      <div
        className="card--body"
      >
        <h3
          title="test-image"
        >
          test-image
        </h3>
        <div
          className="card--details"
        >
          <div
            className="event--id"
            title="Event ID"
          >
            #
            0
          </div>
          <div
            className="event--end"
            title="End Date"
          >
            10-2022
          </div>
        </div>
      </div>
    </div>
  </a>
</div>
`;
